stdlib/test/collections/test_dict.mojo

```diff
@@ -602,6 +603,44 @@ fn test_dict_setdefault() raises:
     assert_equal(0, other_dict["b"].copy_count)
 
 
+def test_dict_entries_del_count():
+    alias dict_type = Dict[Int, ValueDestructorRecorder]
+    var iter = 0
+    for entries_to_insert in range(16):
+        var x = List[Int]()
+        var y = dict_type()
+        var result = 0
+        var expected_result = 0
+        for i in range(entries_to_insert):
+            expected_result += i
+
+        for i in range(entries_to_insert):
+            y[i] = ValueDestructorRecorder(i, UnsafePointer.address_of(x))
+        assert_equal(len(x), 0)
+
+        for i in range(entries_to_insert):
+            result += y[i].value
+        assert_equal(len(x), 0)
+        __type_of(y).__del__(y^)
+        assert_equal(len(x), entries_to_insert)
+        assert_equal(result, expected_result)
+
+        y = dict_type()
+        x.clear()
+        result = 0
+        for i in range(entries_to_insert):
+            y[i] = ValueDestructorRecorder(i, UnsafePointer.address_of(x))
+        assert_equal(len(x), 0)
+        for i in range(entries_to_insert):
+            result += y.pop(i).value
+        assert_equal(len(x), entries_to_insert)
+        __type_of(y).__del__(y^)
+        assert_equal(len(x), entries_to_insert)
+        assert_equal(result, expected_result)
+        iter += 1
+    assert_equal(iter, 16)
```

Comment:
**Question** What's the value in bumping `iter` and then checking the value here?  `assert_*` are "hard errors" and so the stop would executing if any of them failed before we exhausted all the iterations in the fixed for loop.

Reply:
Yep sorry it was after doing `@parameter loops` for another PR
(to test if the range iterate all the alias types)

Should it be removed for this one ?